src/components/Signup/Signup.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import { AUTH_TOKEN } from "../../constants";
+import { Mutation } from "react-apollo";
+import gql from "graphql-tag";
+import { History, LocationState } from "history";
+
+interface Props {
+  history: History<LocationState>;
+}
+interface State {}
+
+const SIGNUP_MUTATION = gql`
+  mutation SignupMutation(
+    $email: String!
+    $password: String!
+    $firstName: String!
+    $lastName: String!
+  ) {
+    signup(
+      email: $email
+      password: $password
+      firstName: $firstName
+      lastName: $lastName
+    ) {
+      token
+    }
+  }
+`;
+
+export default class Signup extends Component<Props, State> {
+  state = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: ""
+  };
+
+  render() {
+    const { firstName, lastName, email, password } = this.state;
+    return (
+      <div>
+        <h1>Signup</h1>
+        <Form
+          className="form"
+          onSubmit={(e: any) => {
+            e.preventDefault();
+          }}
+        >
+          <Form.Group controlId="formFirstName">
+            <Form.Label>First Name:</Form.Label>
+            <Form.Control
+              onChange={(e: any) =>
+                this.setState({ firstName: e.target.value })
+              }
+              type="text"
+              name="firstName"
+              placeholder="Your first name"
+              value={firstName}
+            />
+          </Form.Group>
+          <Form.Group controlId="formLastName">
+            <Form.Label>Last Name:</Form.Label>
+            <Form.Control
+              onChange={(e: any) => this.setState({ lastName: e.target.value })}
```

Comment:
if you are going to use TypeScript try to avoid using `any` as much as possible, set these to `string` or `number` depending

Reply:
I'm not sure what to do with event because when I set the type for event as object, it says target doesn't exist as a property of event.

Reply:
yeah I know its a pain, you have to define an interface somewhere like 
```
interface Event {
target: {
name: string,
value: string | number
}
}
```

Reply:
Oh, I see! Thank you.